### PR TITLE
Enhance Eventing controller logs

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Enhance Eventing controller logs.

**Related issues**

- https://github.com/kyma-project/kyma/issues/11559

### CODE REVIEW

File: components/eventing-controller/pkg/handlers/utils.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ func getQos(qosStr string) (types.Qos, error) {
 func getInternalView4Ev2(subscription *eventingv1alpha1.Subscription, apiRule *apigatewayv1alpha1.APIRule, defaultWebhookAuth *types.WebhookAuth, defaultProtocolSettings *eventingv1alpha1.ProtocolSettings, defaultNamespace string) (*types.Subscription, error) {
 	emsSubscription, err := getDefaultSubscription(defaultProtocolSettings)
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to apply default protocol settings")
+		return nil, errors.Wrap(err, "apply default protocol settings failed")
 	}
```
Comment (pxsalehi) [inline components/eventing-controller/pkg/handlers/utils.go:84]: Just a last comment on these kind of message changes. I think the previous messages are grammatically correct. Why have you decided to make these changes? Is it some sort of logging convention?

File: components/eventing-controller/reconciler/backend/reconciler.go [hunk 2]
```diff
@@ -97,7 +97,6 @@ func (r *Reconciler) Reconcile(ctx context.Context, _ ctrl.Request) (ctrl.Result
 	}); err != nil {
 		return ctrl.Result{}, err
 	}
-	r.namedLogger().Debugw("found secrets with BEB label", "count", len(secretList.Items))
```
Comment (pxsalehi) [inline components/eventing-controller/reconciler/backend/reconciler.go:100]: Why are we removing this? It is a debug message and kind of useful.

File: components/eventing-controller/pkg/commander/beb/beb.go [hunk 3]
```diff
@@ -113,75 +117,64 @@ func (c *Commander) Stop() error {
...
+	return cleanup(c.backend, dynamicClient, c.namedLogger())
+}
+
+func (c *Commander) namedLogger() *zap.SugaredLogger {
+	return c.logger.WithContext().Named(commanderName)
 }
 
 // cleanup removes all created BEB artifacts.
-func cleanup(backend handlers.MessagingBackend, dynamicClient dynamic.Interface) error {
+func cleanup(backend handlers.MessagingBackend, dynamicClient dynamic.Interface, logger *zap.SugaredLogger) error {
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/commander/beb/beb.go:128]: We are using the zap logger which is returned by the kyma unified logger.

File: components/eventing-controller/pkg/commander/beb/beb.go [hunk 4]
```diff
@@ -113,75 +117,64 @@ func (c *Commander) Stop() error {
...
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
-	logger := ctrl.Log.WithName("eventing-controller-beb-cleaner").WithName("Subscription")
+
 	var bebBackend *handlers.Beb
 	var ok bool
-	isCleanupSuccessful := true
 	if bebBackend, ok = backend.(*handlers.Beb); !ok {
-		isCleanupSuccessful = false
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/commander/beb/beb.go:128]: Yes, it is moved down where it is used.

File: components/eventing-controller/pkg/handlers/nats.go [hunk 5]
```diff
@@ -87,77 +82,88 @@ func newCloudeventClient(config env.NatsConfig) (cev2.Client, error) {
...
-		callback := n.getCallback(sub.Spec.Sink)
-
 		if n.connection.Status() != nats.CONNECTED {
-			n.log.Info("connection to Nats", "status", fmt.Sprintf("%v", n.connection.Status()))
-			initializeErr := n.Initialize(env.Config{})
-			if initializeErr != nil {
-				n.log.Error(initializeErr, "failed to reset NATS connection")
-				return false, initializeErr
+			if err := n.Initialize(env.Config{}); err != nil {
+				log.Errorw("reset NATS connection failed", "status", n.connection.Stats(), "error", err)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/nats.go:110]: I'll leave it like this to be consistent with other log messages which are in the present tense.

File: components/eventing-controller/pkg/handlers/nats.go [hunk 6]
```diff
@@ -87,77 +82,88 @@ func newCloudeventClient(config env.NatsConfig) (cev2.Client, error) {
...
 			}
 		}
 
-		natsSub, subscribeErr := n.connection.Subscribe(subject, callback)
-		if subscribeErr != nil {
-			n.log.Error(subscribeErr, "failed to create a Nats subscription")
-			return false, subscribeErr
+		callback := n.getCallback(sub.Spec.Sink)
+		if natsSub, err := n.connection.Subscribe(subject, callback); err != nil {
+			log.Errorw("create NATS subscription failed", "error", err)
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/nats.go:117]: I'll leave it like this to be consistent with other log messages which are in the present tense.

File: components/eventing-controller/pkg/handlers/utils.go (hunk 1 above)
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/utils.go:84]: As discussed, we'll leave it like that to be consistent with other log messages.

File: components/eventing-controller/reconciler/backend/reconciler.go (hunk 2 above)
Comment (marcobebway) [reply components/eventing-controller/reconciler/backend/reconciler.go:100]: This logs creates a lot of noise even in debug mode. Instead, if we found more than one secret with the same label we'll print the found secrets count as part of the context (please check this [commit](https://github.com/kyma-project/kyma/pull/11651/commits/22784ae6b998fd677ecbeab39041f1d69c6bb29e)).

File: components/eventing-controller/reconciler/subscription/reconciler.go [hunk 7]
```diff
@@ -180,7 +177,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	// Save the subscription status if it was changed
 	if statusChanged {
 		if err := r.Status().Update(ctx, desiredSubscription); err != nil {
-			log.Error(err, "Update subscription status failed")
+			log.Errorw("update subscription status failed", "error", err)
```
Comment (sayanh) [inline components/eventing-controller/reconciler/subscription/reconciler.go:180]: Do you wanna add a reconciled successfully log with the subscription name?

File: components/eventing-controller/pkg/handlers/eventtype/parse.go [hunk 8]
```diff
@@ -13,7 +13,7 @@ import (
 // constraint: the application segment in the input event-type should not contain "."
 func parse(eventType, prefix string) (string, string, string, error) {
 	if !strings.HasPrefix(eventType, prefix) {
-		return "", "", "", errors.New(fmt.Sprintf("failed to parse event-type [%s], prefix not found [%s]", eventType, prefix))
+		return "", "", "", errors.New(fmt.Sprintf("parse event-type [%s] failed, prefix not found [%s]", eventType, prefix))
```
[the remark this answers is not in the document]
Comment (marcobebway) [reply components/eventing-controller/pkg/handlers/eventtype/parse.go]: Done.

File: components/eventing-controller/reconciler/subscription/reconciler.go (hunk 7 above)
Comment (marcobebway) [reply components/eventing-controller/reconciler/subscription/reconciler.go:180]: Added.

**Review by sayanh — COMMENTED** [review]
lgtm

**Review by pxsalehi — APPROVED** [review]

**Pull request merged**